scripts/prepdocslib/integratedvectorizerstrategy.py

```diff
@@ -0,0 +1,189 @@
+        self.use_acls = use_acls
+        self.category = category
+
+    async def createEmbeddingSkill(self, index_name: str):
+        skillset_name = f"{index_name}-skillset"
+
+        split_skill = SplitSkill(
+            description="Split skill to chunk documents",
+            text_split_mode="pages",
+            context="/document",
+            maximum_page_length=2048,
+            page_overlap_length=20,
+            inputs=[
+                InputFieldMappingEntry(name="text", source="/document/content"),
+            ],
+            outputs=[OutputFieldMappingEntry(name="textItems", target_name="pages")],
+        )
+
+        if self.embeddings is None:
+            raise ValueError("Expecting Azure Open AI instance")
+
+        embedding_skill = AzureOpenAIEmbeddingSkill(
+            description="Skill to generate embeddings via Azure OpenAI",
+            context="/document/pages/*",
+            resource_uri=f"https://{self.embeddings.open_ai_service}.openai.azure.com",
+            deployment_id=self.embeddings.open_ai_deployment,
+            auth_identity=None,
+            inputs=[
+                InputFieldMappingEntry(name="text", source="/document/pages/*"),
+            ],
+            outputs=[OutputFieldMappingEntry(name="embedding", target_name="vector")],
+        )
+
+        index_projections = SearchIndexerIndexProjections(
+            selectors=[
+                SearchIndexerIndexProjectionSelector(
+                    target_index_name=index_name,
+                    parent_key_field_name="parent_id",
+                    source_context="/document/pages/*",
+                    mappings=[
+                        InputFieldMappingEntry(name="content", source="/document/pages/*"),
+                        InputFieldMappingEntry(name="embedding", source="/document/pages/*/vector"),
+                        InputFieldMappingEntry(name="sourcepage", source="/document/metadata_storage_name"),
+                    ],
+                ),
+            ],
+            parameters=SearchIndexerIndexProjectionsParameters(
+                projection_mode=IndexProjectionMode.SKIP_INDEXING_PARENT_DOCUMENTS
+            ),
+        )
+
+        skillset = SearchIndexerSkillset(
+            name=skillset_name,
+            description="Skillset to chunk documents and generating embeddings",
+            skills=[split_skill, embedding_skill],
+            index_projections=index_projections,
+        )
+
+        return skillset
+
+    async def setup(self, search_info: SearchInfo):
+        search_manager = SearchManager(
+            search_info,
+            self.search_analyzer_name,
+            self.use_acls,
+            True,
+            self.embeddings,
+            search_images=False,
+        )
+
+        if self.embeddings is None:
+            raise ValueError("Expecting Azure Open AI instance")
+
+        await search_manager.create_index(
+            [
```

Comment:
I suggest using the keyword arg here, vectorizers=, since this is an optional param.